apps/checkout-app/pages/customization.tsx

```diff
@@ -37,12 +43,59 @@ const Customization = () => {
     router.back();
   };
 
-  const handleSubmit = (data: CustomizationSettingsValues) => {
+  const uploadSettingsFiles = async (
+    data: CustomizationSettingsValues,
+    dataFiles?: CustomizationSettingsFiles
+  ) => {
+    if (!dataFiles) {
+      return data;
+    }
+
+    return await reduce(
```

Comment:
This part be done better:
- raw reduce inside of reduce is hard to get 
- looking at the length of this function, it probably should be a util function, possibly two
- i don't think we should `await` noun-based variables. So we're awaiting fileUpload (that makes sense). What does it mean to await settings or settings urls?
- I'd much rather see the Promise.resolve done first, then put as reduce's base object. It'd be clearer what it actually is. Sometimes it's better to make it a bit longer but keep some of the stuff named as consts

Reply:
I updated it a bit + also added `Promise.all` to concurrently send files.